src/borg/archive.py

```diff
@@ -680,43 +680,46 @@ def restore_attrs(self, path, item, symlink=False, fd=None):
         gid = item.gid if gid is None else gid
         # This code is a bit of a mess due to os specific differences
         try:
+            try:
```

Comment:
AttributeError means that the method is not available in os for win32 platform.
There are only calls to os methods in the try except block. The first will fail on win32.

We can change the block to a if(  sys.platform.startswith( ("win32", ) or to a flag in platform, if this is more convinient

However such constructs that execute over all platforms should be better in platform code.
But this needs a overall concept and reorganize the platform code itself (even when it is only because of the circular refs)

Reply:
Yes, first will fail. But does that mean you don't want the others either?

Reply:
If first fails this means e.g. Windows windows this attributes (dates rights ect.) are set in a different way.
So Asking for IsWin32 is the right way for now.
This isn't the last touch here.

Reply:
Done